smarts/core/scenario.py

```diff
@@ -516,60 +519,29 @@ def _discover_bubbles(scenario_root):
             bubbles = pickle.load(f)
             return bubbles
 
-    def set_ego_missions(self, ego_mission):
-        self._missions.update(ego_mission)
-
-    @cached_property
-    def traffic_history_lane_width(self):
-        histories_db = sqlite3.connect(self._traffic_history)
-        cur = histories_db.cursor()
-        cur.execute("SELECT value FROM Spec where key='map_net.lane_width'")
-        row = cur.fetchone()
-        cur.close()
-        histories_db.close()
-        return float(row[0]) if row else None
-
-    @cached_property
-    def traffic_history_target_speed(self):
-        histories_db = sqlite3.connect(self._traffic_history)
-        cur = histories_db.cursor()
-        cur.execute("SELECT value FROM Spec where key='speed_limit_mps'")
-        row = cur.fetchone()
-        cur.close()
-        histories_db.close()
-        return float(row[0]) if row else None
+    def set_ego_missions(self, ego_missions: dict):
+        self._missions = ego_missions
 
     def discover_missions_of_traffic_histories(self, vehicle_missions={}):
-        histories_db = sqlite3.connect(self._traffic_history)
-        # For now, limit agent missions to just cars (V.type = 2)
-        st_query = """SELECT T.vehicle_id, min(T.sim_time)
-            FROM Trajectory AS T INNER JOIN Vehicle AS V ON T.vehicle_id=V.id
-            WHERE V.type = 2
-            GROUP BY vehicle_id"""
-        p_query = "SELECT position_x, position_y, heading_rad FROM Trajectory WHERE vehicle_id = ? and sim_time = ?"
         map_offset = self._road_network.net_offset
-        st_cur = histories_db.cursor()
-        for row in st_cur.execute(st_query):
-            vid = str(row[0])
+        for row in self._traffic_history.first_seen_times():
             start_time = float(row[1])
-            p_cur = histories_db.cursor()
-            vrow = p_cur.execute(p_query, (int(vid), start_time)).fetchone()
-            assert vrow
-            pos_x, pos_y, heading = vrow
+            pph = self._traffic_history.vehicle_pose_at_time(row[0], start_time)
+            assert pph
+            pos_x, pos_y, heading = pph
+            vid = str(row[0])
```

Comment:
```suggestion
            v_id = str(row[0])
```

Reply:
I'm going to go ahead and make this change, although I want to push back slightly and say that I don't think we _always_ need to use `v_id` for a vehicle id everywhere in the code,  just in functions and/or files where it's _already_ being called `v_id`.